### DISCUSSION TITLE
SQLite support

### DISCUSSION THREAD

**Discussion by dickermoshe** [serverpod/serverpod#1977] [General]
I assume Serverpod mini doesn't have a database because setting up PG for a simple server would be annoying.
Django avoids this by having SQLite by default.
How tied is Serverpod to PG, theoretically can sqlite support be added?

**Comment by SandPod** [comment]
Hey!

Serverpod internally has a PG adapter implemented; architecturally speaking it is prepared for connection to other SQL databases.
If we would like to use a different query language such as a document database there would be more problems as your expressions depend on a relational database.

But adding an SQL lite adapter theoretically speaking shouldn't be too much work.

**Reply by dkbast** [reply]
This could be very useful to add offline (first) support for apps - if the generated client could also handle "local persistence" that would be a true gamechanger.

**Comment by dickermoshe** [comment]
@SandPod 
The reason I'm bringing this up, is that serverpod mini was created to make serverpod easier to use without setting up a database.
Coming from Django, they do this by using sqlite by default. You then can switch to PG in prod.
Seems that it would be smarter in the long term to do something like that , rather than maintaining 2 different code bases.

**Reply by SandPod** [reply]
Sounds smart to use sqlite in a dev environment.

Mini was made since databases (among other things) in general might be a difficult concept understand. SQLite probably doesn't help in that case. But it would be really neat to make the adapter swappable depending on environment (if we add support for SQLite).
